pkg/virt-handler/rest/console.go

```diff
@@ -138,20 +138,40 @@ func (t *ConsoleHandler) USBRedirHandler(request *restful.Request, response *res
 	t.stream(vmi, request, response, unixSocketDialer(vmi, unixSocketPath), stopChan)
 }
 
+func (t *ConsoleHandler) vncInUse(uid types.UID) bool {
+	t.vncLock.Lock()
+	defer t.vncLock.Unlock()
+	_, inUse := t.vncStopChans[uid]
+	return inUse
+}
+
 func (t *ConsoleHandler) VNCHandler(request *restful.Request, response *restful.Response) {
 	vmi, code, err := getVMI(request, t.vmiStore)
 	if err != nil || vmi == nil {
 		log.Log.Reason(err).Error(failedRetrieveVMI)
 		response.WriteError(code, err)
 		return
 	}
+
+	uid := vmi.GetUID()
+	preserveSessionParam, err := strconv.ParseBool(request.QueryParameter("preserveSession"))
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Warningf("VNC's query parameter preserveSession")
+	}
+
+	if t.vncInUse(uid) && preserveSessionParam {
+		err = fmt.Errorf("%s", "Active VNC connection. Request denied.")
+		log.Log.Object(vmi).Reason(err).Error("Already has an active connection. Preserve it.")
+		response.WriteError(http.StatusNotAcceptable, err)
```

Comment:
Not sure about the returned status code. It's not a client error. What about 503 instead?

Reply:
Probably related to https://github.com/kubevirt/kubevirt/issues/9463

It could be because these APIs are layered behind the Kubernetes API